fairmq/shmem/UnmanagedRegion.h

```diff
@@ -245,12 +269,18 @@ struct UnmanagedRegion
     std::vector<RegionBlock> fBlocksToFree;
     const std::size_t fAckBunchSize = 256;
     std::unique_ptr<boost::interprocess::message_queue> fQueue;
+    std::unique_ptr<boost::interprocess::managed_shared_memory> fRefCountSegment;
+    std::unique_ptr<RefCountPool> fRefCountPool;
 
     std::thread fAcksReceiver;
     std::thread fAcksSender;
     RegionCallback fCallback;
     RegionBulkCallback fBulkCallback;
 
+    static std::string MakeSegmentName(const std::string& shmId, std::string_view segment, int regionIndex) {
+        return tools::ToString("fmq_", shmId, "_", segment, "_", regionIndex);
+    }
```

Comment:
I realize now, you generate these segment names in many more places. I guess, I would then move this helper to `Common.h` and use it everywhere for the purpose of deduplicating the segment name logic to a single place.

Reply:
Sure

Reply:
ah, and note I sneaked in `std::string_view` for the second argument, not sure if we include the `<string_view>` header already.

Reply:
actually, I'm not sure what the impact is of replacing a simple constructor arguments with our ToString tool is, since the tool constructs a stringstream, among other things. For this contructor I don't care since it is called once. But some of the other usage (in the Monitor) is potentially called in a loop, so I would rather keep it minimal there.

Reply:
Or you could not use `tools::ToString` in the helper, and use the `+` operators just to be sure. My point would be to not duplicate the naming logic and the constant prefix etc. Easy to mess up/overlook in future refactorings etc.

Reply:
Side note: In general, I would vote to adopt libfmt and modernize those types of string generation - I think we have a couple of places that could benefit from this. That's of course out of scope of this PR. The other day I had a look at [`std::print`](https://en.cppreference.com/w/cpp/io/print) (C++23) and realized it is not even implemented in GCC yet. So, this will be many years in the future still, so an adoption of libfmt seems worth.